caringcaribou/modules/uds.py

```diff
@@ -714,6 +816,212 @@ def __dump_dids_wrapper(args):
+
+                print("\nEnumerating Services:\n")
+
+                found_services = service_discovery(client_id, server_id, timeout)
+
+                print("\nIdentified services:\n")
+
+                # Print available services result table
+                for service_id in found_services:
+                    service_name = UDS_SERVICE_NAMES.get(service_id, "Unknown service")
+                    print("Supported service 0x{0:02x}: {1}"
+                        .format(service_id, service_name))  
+                
+                print("\n")
+                    
+                dump_dids(client_id, server_id, timeout, min_did, max_did, print_results)
+
+                if 16 in found_services:
+
+                    print("\nEnumerating Diagnostic Session Control Service:\n")
+
+                    found_subservices = []
+                    subservice_status = []
+
+                    for i in range(1,256):    
+
+                        extended_session(client_id, server_id, 1)
+
+                        #time.sleep(0.1)
+
+                        response = extended_session(client_id, server_id, i)
+
+                        print("\rProbing diagnostic session control sub-service 0x{0:02x}".format(i), end="")
+
+                        if response is None:
+                            # No response received
+                            continue
+
+                        # Parse response
+                        if len(response) >= 3:
+                            response_id = response[0]
+                            response_service_id = response[1]
+                            status = response[2]
+                            if Iso14229_1.is_positive_response(response):
+                                found_subservices.append(i)
+                                subservice_status.append(0x00)
+                            elif response_id == Constants.NR_SI and response_service_id == 0x10 and status != NegativeResponseCodes.SUB_FUNCTION_NOT_SUPPORTED:
+                                # Any other response than "service not supported" counts
+                                found_subservices.append(i)
+                                subservice_status.append(response_service_id)
+
+                        time.sleep(timeout)
+
+                    # Print results
+                    if len(found_subservices) == 0:
+                        print("\nNo Diagnostic Session Control Sub-Services were discovered\n", end=' ')
+                    else:
+                        print("\n")
+                        print("\nDiscovered Diagnostic Session Control Sub-Services:\n", end=' ')
+                        for subservice_id in found_subservices:
+                            nrc_description = NRC_NAMES.get(subservice_status[found_subservices.index(subservice_id)])
+                            print("\n0x{0:02x} : {1}".format(subservice_id, nrc_description), end=' ')
+                
+                if 17 in found_services:
```

Comment:
Magic number - this should probably reference ECU Reset (set to `0x11`) through a variable instead for improved readability.